selfservice/strategy/password/login_test.go

```diff
@@ -864,4 +860,168 @@ func TestCompleteLogin(t *testing.T) {
 			false, true, http.StatusOK, redirTS.URL)
 		assert.Equal(t, identifier, gjson.Get(body, "identity.traits.subject").String(), "%s", body)
 	})
+
+	t.Run("suite=password migration hook", func(t *testing.T) {
+		ctx := context.Background()
+
+		passwordDB := make(map[string]string)
```

Comment:
What I would like instead is to have a channel here that you send a response (function) to instead of this custom logic. Then the server is super simple, it just receives the response from the channel and writes it, and the test setup is also very easy to understand without reading the server code.

```suggestion
		passwordDB := make(chan func(context.Context, http.ResponseWriter), 1)
		writeStatus := func(status int) func(context.Context, http.ResponseWriter) {
			return func(_ context.Context, w http.ResponseWriter) {
				w.WriteHeader(status)
			}
		}
		expectedResponseBody := `{"status":"password_match"}`
		writeBodyOK := func(body string) func(context.Context, http.ResponseWriter) 
			return func(_ context.Context, w http.ResponseWriter) {
				w.WriteHeader(http.StatusOK)
				_, _ = io.WriteString(w, body)
			}
		}
```